query/src/main/scala/filodb/query/PromCirceSupport.scala

```diff
@@ -41,29 +43,39 @@ object PromCirceSupport {
       }
     }
 
-  implicit val decodeFoo: Decoder[DataSampl] = new Decoder[DataSampl] {
-    final def apply(c: HCursor): Decoder.Result[DataSampl] = {
+  implicit val decodeMetadataSampl: Decoder[MetadataSampl] = new Decoder[MetadataSampl] {
+    def apply(c: HCursor): Decoder.Result[MetadataSampl] = {
       c.downField("cardinality").focus match {
         case Some(_) =>
           // TODO: Not the best way to find if this is a label cardinality response, is there a better way?
-          for {metric     <- c.get[Map[String, String]]("metric")
-               card      <- c.get[Seq[Map[String, String]]]("cardinality")
-               } yield LabelCardinalitySampl(metric, card)
+          for {
+            metric <- c.get[Map[String, String]]("metric")
+            card <- c.get[Seq[Map[String, String]]]("cardinality")
+          } yield LabelCardinalitySampl(metric, card)
 
-        case None              =>
-          val tsResult = c.downArray.as[Long]
-          val rightCurs = c.downArray.right
-          if (rightCurs.focus.get.isObject) {
-            for { timestamp <- tsResult
-                  buckets   <- rightCurs.as[Map[String, Double]] } yield {
-              HistSampl(timestamp, buckets)
-            }
-          } else {
-            for { timestamp <- tsResult
-                  value     <- rightCurs.as[String] } yield {
-              Sampl(timestamp, value.toDouble)
-            }
-          }
+        case None =>
+          if (c.value.isString)
+            for { label <- c.as[String] } yield LabelSampl(label)
+          else
+            for { metadataMap <- c.as[Map[String, String]] } yield MetadataMapSampl(metadataMap)
+      }
+    }
+  }
+
+  implicit val decodeFoo: Decoder[DataSampl] = new Decoder[DataSampl] {
```

Comment:
do we really have to use this variable name as `decodeFoo` ?

Reply:
renamed it